src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp

```diff
@@ -753,17 +819,26 @@ RelationalDBInterfaceSqliteImp::getLedgerInfoByHash(uint256 const& ledgerHash)
     }
 
     /* else use shard databases */
-    std::optional<LedgerInfo> res;
-    iterateLedgerBack({}, [&](soci::session& session, std::uint32_t index) {
-        if (auto info = ripple::getLedgerInfoByHash(session, ledgerHash, j_))
+    if (app_.getShardStore())
```

Comment:
Since the shard store is called below, consider using...
```if (auto shardStore = app_.getShardStore())```

Reply:
Done